include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h

```diff
@@ -0,0 +1,242 @@
+                  std::string const &                            field,
+                  MPI_Comm const &                               mpi_comm);
+
+  void
+  fill_matrix_free_data(MatrixFreeData<dim, Number> & matrix_free_data) const;
+
+  /**
+   * Call this setup() function if the dealii::MatrixFree object can be set up by the present class.
+   */
+  void
+  setup();
+
+  /**
+   * Call this setup() function if the dealii::MatrixFree object needs to be created outside this
+   * class. The typical use case would be multiphysics-coupling with one MatrixFree object handed
+   * over to several single-field solvers. Another typical use case is the use of an ALE
+   * formulation.
+   */
+  void
+  setup(std::shared_ptr<dealii::MatrixFree<dim, Number> const> matrix_free_in,
+        std::shared_ptr<MatrixFreeData<dim, Number> const>     matrix_free_data_in);
+
+  /*
+   * Getters and setters.
+   */
+  dealii::MatrixFree<dim, Number> const &
+  get_matrix_free() const;
+
+  std::string
+  get_dof_name_pressure() const;
+
+  unsigned int
+  get_dof_index_pressure() const;
+
+  std::string
+  get_dof_name_velocity() const;
+
+  unsigned int
+  get_dof_index_velocity() const;
+
+  unsigned int
+  get_quad_index_pressure() const;
+
+  unsigned int
+  get_quad_index_velocity() const;
+
+  unsigned int
+  get_quad_index_pressure_velocity() const;
+
+  std::shared_ptr<dealii::Mapping<dim> const>
+  get_mapping() const;
+
+  dealii::FiniteElement<dim> const &
+  get_fe_p() const;
+
+  dealii::FiniteElement<dim> const &
+  get_fe_u() const;
+
+  dealii::DoFHandler<dim> const &
+  get_dof_handler_p() const;
+
+  dealii::DoFHandler<dim> const &
+  get_dof_handler_u() const;
+
+  dealii::AffineConstraints<Number> const &
+  get_constraint_p() const;
+
+  dealii::AffineConstraints<Number> const &
+  get_constraint_u() const;
+
+  dealii::types::global_dof_index
+  get_number_of_dofs() const;
+
+  /*
+   * Initialization of vectors.
+   */
+  void
+  initialize_dof_vector(BlockVectorType & dst) const final;
+
+  /*
+   * Prescribe initial conditions using a specified analytical/initial solution function.
+   */
+  void
+  prescribe_initial_conditions(BlockVectorType & dst, double const time) const final;
+
+  /*
+   *  This function is used in case of explicit time integration:
+   *  This function evaluates the right-hand side operator, the
+   *  convective and viscous terms (subsequently multiplied by -1.0 in order
+   *  to shift these terms to the right-hand side of the equations)
+   *  and finally applies the inverse mass operator.
+   */
+  void
+  evaluate(BlockVectorType & dst, BlockVectorType const & src, double const time) const final;
+
+  /*
+   * Operators.
+   */
+
+  // acoustic operator
+  void
+  evaluate_acoustic_operator(BlockVectorType &       dst,
+                             BlockVectorType const & src,
+                             double const            time) const;
+
+  // inverse velocity mass operator
```

Comment:
why velocity?

Reply:
this is a typo

Reply:
[b81e2b0](https://github.com/exadg/exadg/pull/592/commits/b81e2b087d1b4b947127493ddf050e15bb9f59a4)